src/main/java/io/pravega/connectors/hadoop/PravegaOutputFormat.java

```diff
@@ -0,0 +1,176 @@
+        final String serializerClassName = Optional.ofNullable(conf.get(PravegaConfig.OUTPUT_SERIALIZER)).orElseThrow(() ->
+                new IOException("The event serializer must be configured (" + PravegaConfig.OUTPUT_SERIALIZER + ")"));
+
+        final String eventRouterClassName = conf.get(PravegaConfig.OUTPUT_EVENT_ROUTER);
+        PravegaEventRouter<V> pravegaEventRouter = getEventRouter(eventRouterClassName);
+
+        return new PravegaOutputRecordWriter<>(getPravegaWriter(scopeName, streamName, controllerURI, serializerClassName), pravegaEventRouter);
+    }
+
+    @Override
+    public void checkOutputSpecs(JobContext jobContext) throws IOException, InterruptedException {
+    }
+
+    @Override
+    public OutputCommitter getOutputCommitter(TaskAttemptContext taskAttemptContext) throws IOException, InterruptedException {
+        return new PravegaOutputCommitter(new Path("/tmp/" + taskAttemptContext.getTaskAttemptID().getJobID().toString()), taskAttemptContext);
+    }
+
+    private PravegaEventRouter<V> getEventRouter(String eventRouterClassName) throws IOException {
+        PravegaEventRouter<V> serializer = null;
+        if (eventRouterClassName != null) {
+            try {
+                Class<?> serializerClass = Class.forName(eventRouterClassName);
+                serializer = (PravegaEventRouter<V>) serializerClass.newInstance();
+            } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+                log.error("Exception when creating serializer: {}", e);
+                throw new IOException(
+                        "Unable to create the serializer for event router (" + eventRouterClassName + ")", e);
+            }
+        }
+        return serializer;
+    }
+
+    private EventStreamWriter<V> getPravegaWriter(
+            String scopeName,
+            String streamName,
+            URI controllerURI,
+            String serializerClassName) throws IOException {
+        ClientFactory clientFactory = (externalClientFactory != null) ? externalClientFactory : ClientFactory.withScope(scopeName, controllerURI);
+
+        Serializer serializer;
+
+        try {
+            Class<?> serializerClass = Class.forName(serializerClassName);
+            serializer = (Serializer<V>) serializerClass.newInstance();
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            log.error("Exception when creating serializer: {}", e);
+            throw new IOException(
+                    "Unable to create the event serializer (" + serializerClassName + ")", e);
+        }
+
+        EventStreamWriter<V> writer = clientFactory.createEventWriter(streamName, serializer, EventWriterConfig.builder()
+                .transactionTimeoutTime(DEFAULT_TXN_TIMEOUT_MS)
+                .build());
+
+        return writer;
+    }
+
+    class PravegaOutputCommitter extends FileOutputCommitter {
```

Comment:
I don't understand why `FileOutputCommitter` would be useful here. 
https://github.com/apache/hadoop/blob/trunk/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/output/FileOutputCommitter.java

Reply:
My understanding is that the output committer is used to keep track of the task progress. Do you see a problem in using `FileOutputCommitter`?